### PR TITLE
Issue #SC-1808: Denormalize user metadata in the job

### PR DESCRIPTION
**Pull request by utk14**
The PR contains the changes into the user-cache-updater job in which we are enhancing the job for denormalizing the user metadata wherein we fetch the data from the cassandra table and update the data into cache in hashmap format. 
1. User information from User table
2. Information required for reports based on the custodian user or state user.

### CODE REVIEW

File: data-pipeline-flink/user-cache-updater/src/main/scala/org/sunbird/dp/usercache/functions/UserCacheUpdaterFunction.scala [hunk 1]
```diff
@@ -152,9 +152,121 @@ class UserCacheUpdaterFunction(config: UserCacheUpdaterConfig)(implicit val mapT
...
     })
     result
   }
 
+  /**
+    * It Will return the rootOrgId to filter out custodian users and state users
+    * @param metrics
+    * @return custodianRootOrgId
+    */
+  def getCustodianRootOrgId(metrics: Metrics): String = {
```
Comment (anandp504) [inline data-pipeline-flink/user-cache-updater/src/main/scala/org/sunbird/dp/usercache/functions/UserCacheUpdaterFunction.scala:166]: This value needs to be cached if the CustodianOrgId doesn't change. We should not be making so many calls when it returns the same value.

File: data-pipeline-flink/user-cache-updater/src/main/scala/org/sunbird/dp/usercache/functions/UserCacheUpdaterFunction.scala [hunk 2]
```diff
@@ -152,9 +152,121 @@ class UserCacheUpdaterFunction(config: UserCacheUpdaterConfig)(implicit val mapT
...
+      metrics)
+    custRootOrgId.get(0).getString(2)
+  }
+
+  /**
+    *
+    * @param userDetails, custRootOrgId
+    * @return Map having isCustodianUser as true/false
+    */
+  def getIsCustodianUser(userDetails: mutable.Map[String, AnyRef], custRootOrgId: String): mutable.Map[String, AnyRef] ={
```
Comment (anandp504) [inline data-pipeline-flink/user-cache-updater/src/main/scala/org/sunbird/dp/usercache/functions/UserCacheUpdaterFunction.scala:180]: I am assuming the function is used to check if a user belongs to a custodian org. The function should be named appropriately. Also, we shouldn't be passing around a mutable Map to add a boolean value to the Map. The value needs to be updated in the calling function.

File: data-pipeline-flink/user-cache-updater/src/main/scala/org/sunbird/dp/usercache/functions/UserCacheUpdaterFunction.scala (hunk 1 above)
Comment (utk14) [reply data-pipeline-flink/user-cache-updater/src/main/scala/org/sunbird/dp/usercache/functions/UserCacheUpdaterFunction.scala:166]: Done

File: data-pipeline-flink/user-cache-updater/src/main/scala/org/sunbird/dp/usercache/functions/UserCacheUpdaterFunction.scala (hunk 2 above)
Comment (utk14) [reply data-pipeline-flink/user-cache-updater/src/main/scala/org/sunbird/dp/usercache/functions/UserCacheUpdaterFunction.scala:180]: Done

**Comment by utk14** [comment]
A new PR will be raised

**Pull request closed without merging**